Firestore/Source/API/FIRQuerySnapshot.mm

```diff
@@ -129,13 +137,10 @@ - (NSInteger)count {
 
     NSMutableArray<FIRQueryDocumentSnapshot *> *result = [NSMutableArray array];
     for (FSTDocument *document in documentSet.documentEnumerator) {
-      [result
-          addObject:[FIRQueryDocumentSnapshot
-                        snapshotWithFirestore:firestore
-                                  documentKey:document.key
-                                     document:document
-                                    fromCache:fromCache
-                             hasPendingWrites:self.snapshot.mutatedKeys.contains(document.key)]];
+      auto apiResult = absl::make_unique<QueryDocumentSnapshot>(
```

Comment:
Reverted this change, so that calling code stays the same. I tried doing a test helper, but I found that there already exists a `FSTTestDocSnapshot` helper that takes different parameters, and distinguishing between the two seemed too hard to be worth it.